packages/flutter_tools/lib/src/version.dart

```diff
@@ -236,21 +236,38 @@ class FlutterVersion {
     final Duration frameworkAge = _clock.now().difference(localFrameworkCommitDate);
     final bool installationSeemsOutdated = frameworkAge > kVersionAgeConsideredUpToDate;
 
-    Future<bool> newerFrameworkVersionAvailable() async {
+    /// Gets whether or not there is a new version of Flutter available on the remote.
+    Future<VersionCheckResult> newerFrameworkVersionAvailable() async {
       final DateTime latestFlutterCommitDate = await _getLatestAvailableFlutterVersion();
 
       if (latestFlutterCommitDate == null)
-        return false;
+        return VersionCheckResult.unknown;
 
-      return latestFlutterCommitDate.isAfter(localFrameworkCommitDate);
+      return latestFlutterCommitDate.isAfter(localFrameworkCommitDate)
+          ? VersionCheckResult.newVersionAvailable
+          : VersionCheckResult.versionIsCurrent;
     }
 
+    // Get whether there's a newer version on the remote. This only goes
+    // to the server if we haven't checked recently so won't happen often.
+    // Note: This must come before the stamp load below because it also reads/writes
+    // the stamp and otherwise one will be stale.
+    final VersionCheckResult remoteVersion = await newerFrameworkVersionAvailable();
```

Comment:
it's kind of weird to have this be done in a nested function instead of just inline.